### DISCUSSION TITLE
How to syncronize two Y-Axis with a Grid

### DISCUSSION THREAD

**Discussion by lanwin** [airbnb/visx#1042] [Q&A]
Hi, I have two Y-Axis with two different linearScales. And now I want to align both axis with the grid and can not find a good way to do so:

![grafik](https://user-images.githubusercontent.com/15905/106590108-0f638680-654d-11eb-9a1e-84b128c91c26.png)

Ive tried to give both the same numTicks but it did not work.

Any suggestions?

**Comment by williaster** [comment]
Hey @lanwin 👋 thanks for checking out `visx`. This is a tricky one, nothing specific to `visx` here so you could potentially look for `d3` solutions as well. Note that `numTicks` is approximate due to the underlying `d3` algorithm and because they are entirely different scales there's no guarantee that even if the count was the same that that would divide the available space in the same way. You'd likely have to do some math yourself to find a common denominator between the axes and use `tickValues` for them to be exact.

Generally I would recommend **against** trying to overlay two y-axes on the same chart, it's often seen as violating "data vis best practices" because they are confusing and often uninterpretable to users. For instance in looking at this screenshot, I cannot tell whether the line maps to the left axis or the right, and same for the bars (often users that insist on using two y-axes try to color match the scales to make this unambiguous). Two y-axes also can be misleading in terms of suggesting that the two axes values are correlated somehow, even though the correlation depends **mostly** on the scales of the two axes.

Instead I recommend vertically stacking two charts on top of each other so they can leverage the same x-scale, while having distinct and interpretable y-axes. Usually the reason you want to overlay them is to share the x-axis, and this approach gives you that.
